test/gtest/ucp/test_ucp_mmap.cc

```diff
@@ -355,18 +358,32 @@ bool test_ucp_mmap::disable_proto() const
     return get_variant_value() == VARIANT_PROTO_DISABLE;
 }
 
-void test_ucp_mmap::expect_same_distance(const ucs_sys_dev_distance_t &dist1,
-                                         const ucs_sys_dev_distance_t &dist2)
+void test_ucp_mmap::check_distance_precision(double rkey_value,
+                                             double topo_value,
+                                             size_t pack_min,
+                                             size_t pack_max)
 {
     /* Expect the implementation to always provide a reasonable precision w.r.t.
      * real-world bandwidth and latency ballpark numbers.
      */
-    EXPECT_NEAR(dist1.bandwidth, dist2.bandwidth, 600e6); /* 600 MBs accuracy */
-    EXPECT_NEAR(dist1.latency, dist2.latency, 20e-9); /* 20 nsec accuracy */
+    double allowed_diff_ratio = 1 - UCS_FP8_PRECISION;
+
+    if (rkey_value == pack_min) {
```

Comment:
it's also not casted

Reply:
In that case we set only exponent part for rkey value, so casting isn't required

Reply:
ok, seems there should be no warnings